Controllers/ApiDataController.cs

```diff
@@ -0,0 +1,56 @@
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Text;
+using System.Net;
+using RestSharp;
+using System.Collections.Generic;
+using whale_spotting.Models.ApiModels;
+using whale_spotting.Models.Database;
+using whale_spotting.Repositories;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace whale_spotting.Controllers
+{
+    [ApiController]
+    [Route("/getapidata")]
+    public class ApiDataController : ControllerBase
+    {     
+        private readonly ISightingsRepo _sightings;
+        public ApiDataController(ISightingsRepo sightings)
+        {
+            _sightings = sightings;
+        }
+
+        [HttpPost("")]
+        public async Task<IActionResult> GetApiData() 
+        {
+            int page = 1;
+            bool Flag = true;
```

Comment:
can we call this something descriptive just for readability like resultsExist or something along those lines that might be better